cmdtests/common.go

```diff
@@ -120,3 +105,147 @@ func (s *IntegrationTmpDirSuite) SetupTest() {
 func (s *IntegrationTmpDirSuite) TearDownTest() {
 	os.RemoveAll(s.TestDir)
 }
+
+type RegressionSuite struct {
+	suite.Suite
+	PrevCmd *Commander
+	CurrCmd *Commander
+}
+
+func NewRegressionSuite(prevBin, currentBin string) RegressionSuite {
+	return RegressionSuite{
+		PrevCmd: &Commander{bin: prevBin},
+		CurrCmd: &Commander{bin: currentBin},
+	}
+}
+
+type SQLOutputTable struct {
+	Data  map[string][]string
+	cols  []string
+	rowsN int
+}
+
+func (s *SQLOutputTable) RequireEqual(o *SQLOutputTable) error {
+	return s.requireEqual(o, false)
+}
+
+func (s *SQLOutputTable) RequireStrictlyEqual(o *SQLOutputTable) error {
+	return s.requireEqual(o, true)
+}
+
+func (s *SQLOutputTable) requireEqual(o *SQLOutputTable, strictEmpty bool) error {
+	if !strictEmpty && s.rowsN == 0 && o.rowsN == 0 {
+		return nil
+	}
+
+	if s.rowsN != o.rowsN {
+		return s.diffErr("rows number", s.rowsN, o.rowsN)
+	}
+
+	if !reflect.DeepEqual(s.cols, o.cols) {
+		return s.diffErr("columns", s.cols, o.cols)
+	}
+
+	var thisRows []string
+	var otherRows []string
+
+	for i := 0; i < s.rowsN; i++ {
+		var thisRow []string
+		var otherRow []string
+
+		for _, c := range s.cols {
+			thisRow = append(thisRow, s.Data[c][i])
+			otherRow = append(otherRow, o.Data[c][i])
+		}
+
+		thisRows = append(thisRows, strings.Join(thisRow, "|"))
+		otherRows = append(otherRows, strings.Join(otherRow, "|"))
+	}
+
+	sort.Strings(thisRows)
+	sort.Strings(otherRows)
+
+	eq := reflect.DeepEqual(thisRows, otherRows)
+	if eq {
+		return nil
+	}
+
+	return s.diffErr("rows", thisRows, otherRows)
+}
+
+func (s *SQLOutputTable) diffErr(what string, this, other interface{}) error {
+	return fmt.Errorf("Different %s:\n- actual:   %v\n- expected: %v",
+		what, this, other)
+}
+
+func AreSQLOutputStrictlyEqual(s1 string, s2 string) error {
+	return ParseSQLOutput(s1).RequireStrictlyEqual(ParseSQLOutput(s2))
+}
+
+func AreSQLOutputEqual(s1 string, s2 string) error {
+	return ParseSQLOutput(s1).RequireEqual(ParseSQLOutput(s2))
```

Comment:
why do you need to parse results? Is there any problem with simple comparing strings?

Reply:
AFAIR it was for the rows ordering, but now I guess that we may force the query to include an `ORDER BY` clause. But I don't know whether in the future we would need to test queries strictly without `ORDER BY` clauses.

Reply:
I don't mind to merge it as is. It just seems like an extra code that we can avoid so I mentioned it.
//cc @carlosms

Reply:
:thinking: it's a shame to ask you to remove code that is already in place, but it's true that right now it's a bit convoluted.

If the only thing thing that prevents us from comparing the strings directly is the `ORDER BY`, I think it's worth it to remove this extra code, to keep the code cleaner and easier to follow.

Reply:
Oh no! I remember now! But I don't know whether we want to still keep it...

There's (was) another reason. When I developed this at some point it was in-between the replacement of the SQL cli, so there was a version using our custom sql cli and another using mysql cli.

In one case the headers were all capitalized and in the other not, and also if the header is composed by multiple words the separator in one case is the space and in the other was a dash. Moreover, in our sql cli the newline was `\n` while on mysql is `\r\n`, but this latter thing is a minimal problem as we can simply replace all the newlines of the buffer.

So if we still want to test versions previous to the sql cli replacement this is needed, but I don't know whether it does make sense to test such older versions.

Reply:
I we were writing it now I would say we could ignore previous versions. Since it's already done maybe we can keep it, but let's add a comment explaining why this is done. Then later we can remove it if we don't use this to test older versions at all.